### ISSUE TITLE
Beginning Rust - fast track your learning

### ISSUE THREAD

**Issue by tcharding** [RustSydney/talks#36]
I recently got a job writing Rust.  I did not know Rust at the time.  This was a good opportunity to try out a language learning method I had floating around my head for some time.  In this presentation I will present the method I have used / am using to fast track my Rust learning.

The method is quite simple.  Pick a selection of algorithms (and code snippets) of increasing complexity that cover parts of the language one wishes to learn.  Write out the algos each day until one can write them without pausing/thinking.  On the surface this is quite trivial and not at all revolutionary.  I hope to inspire people to try this method since it has been very positive for me, also I hope the algos/snippets I have used will be of some use/interest to the group.  I also hope to learn a few ways to improve the method and/or algos by presenting the topic.

Thanks

**Comment by tcharding** [comment]
Seems I cannot add the 'presentation' label.

**Comment by bitshifter** [comment]
Sounds like a good topic, I'll get in touch when we're planning our next meetup. It probably won't be for a few months.

**Comment by tcharding** [comment]
Cool, thanks.

**Comment by tcharding** [comment]
Just wrote a blog post about this.  Linking here so people can see better what this proposed talk will be about.

http://tobin.cc/blog/learning-rust/

**Comment by bitshifter** [comment]
Hi @tcharding , we're planning a meetup on Tuesday the 29th of October. Would you still be interested in giving a talk at that time? If so, what kind of duration were you thinking for your talk? We have a few talk proposals at the moment so I'm trying to find out who is still interested and available initially.

**Comment by tcharding** [comment]
Hi @bitshifter, just saw this.  Yes I am still keen to present this topic if you have a slot.  
Only needs as short slot, 10 or 20 minutes perhaps.  Thanks.

**Comment by bitshifter** [comment]
Hi @tcharding , we've got a couple of speakers but I think we could fit you in if it's a short one. When you have time send me a talk title/summary and optionally a bio for the meetup description.

**Comment by tcharding** [comment]
Great, thanks @bitshifter, will do.

**Comment by bitshifter** [comment]
Event is up https://www.meetup.com/Rust-Sydney/events/265708002/

**Comment by tcharding** [comment]
Learning Rust via Simple Algorithms
===================================

Learning new languages is fun, no doubt.  Before the new language can become a
tool that is comfortably wielded, and after the initial excitement of the new
language has worn off, there is a time where repeatedly looking up how to do
things can become annoying.

In this quick presentation I will share a method I used recently to help come up
to speed with Rust - typing out simple algorithms, the same one each day until
it became rote.  I'll run through the method and share the algorithms I used.

Bio
---

Tobin is a Rust hacker at CoBloX here in Sydney.  He works, along with five
other engineers, on the reference implementation of the open source COMIT
protocol which attempts to solve atomic trustless cross-chain exchange of
assets.  He is new to the crypto space and to Rust, he thoroughly enjoys both.
